### ISSUE TITLE
Add Practice Exercise: `transpose`

### ISSUE THREAD

**Issue by 0xNeshi** [exercism/cairo#241]
Problem specification:
https://github.com/exercism/problem-specifications/tree/main/exercises/transpose
    
TODO:
    
- generate the exercise with `./bin/fetch-configlet && ./bin/configlet create --practice-exercise transpose`
- check if any of the existing tracks (examples below) implement the exercise and you can reuse the implementation:
    - Go: https://github.com/exercism/go/tree/main/exercises/practice
    - Rust: https://github.com/exercism/rust/tree/main/exercises/practice
    - Python: https://github.com/exercism/python/tree/main/exercises/practice
- use exercise's _canonical-data.json_ to see which test cases to implement:
    https://github.com/exercism/problem-specifications/blob/main/exercises/transpose/canonical-data.json

**Comment by Ephraim-nonso** [comment]
Let me take this.

**Comment by shreyasgosavi** [comment]
Hello I am really interested to make my contribution but am new to open-source world. 

Will be happy to take the task, can anyone just give overview of what is expected?  assign it to me

**Comment by 0xNeshi** [comment]
Hi @shreyasgosavi!

Thank you for wanting to contribute to open-source :rocket:

This particular issue is already taken, but you're free to pick any of the available ones!

May I suggest one of the easier ones, for example to [implement the flatten-array exercise](https://github.com/exercism/cairo/issues/180)?
If you'd like to work on this, just comment on that (or any other issue) and I'll assign it to you :)

After forking the repository, simply follow the instructions in the issue description and most of the code will be generated for you, which is awesome!

You can take a look at other closed PRs that added existing practice exercises to see what you'll need to do, e.g. [this is a good example that implemented `isbn-verifier`](https://github.com/exercism/cairo/pull/309/files). 
You'll only need to implement 3 things: 
1. An example solution (_example.cairo_).
2. A scaffold solution (_lib.cairo_).
3. Tests (_<exercise-name>.cairo_).

For the example solution, you can use any of the other track's implementations, e.g. for this exercise [the Python's implementation](https://github.com/exercism/python/blob/main/exercises/practice/flatten-array/.meta/example.py) is very straightforward! 

You'll need to add test cases, for which inputs and expected results are already provided, check out https://github.com/exercism/problem-specifications/blob/main/exercises/flatten-array/canonical-data.json

Make sure to read the [README file](https://github.com/exercism/cairo/blob/main/README.md) for instructions on testing, formatting etc.!

**Comment by 0xNeshi** [comment]
@Ephraim-nonso are you working on this?
If not, I can reassign the issue

**Comment by shreyasgosavi** [comment]
Hi Nenad,

Would really like to work on the one you mentioned, will add a comment on
that.

Thank you for the response & your time!!

Regards,
Shreyas Gosavi

**Comment by Ephraim-nonso** [comment]
> @Ephraim-nonso are you working on this?
> 
> If not, I can reassign the issue

Yes. I'm still working on this issue. Apologies it's taking much more time than usual.

**Comment by 0xNeshi** [comment]
@Ephraim-nonso no problem, let me know if I can be of help

**Comment by Ephraim-nonso** [comment]
> @Ephraim-nonso no problem, feel free to let me know if I can help

I'd revert latest today on this.

**Comment by shreyasgosavi** [comment]
Ok you can assign me some other task if this is already taken ...

**Comment by 0xNeshi** [comment]
@shreyasgosavi I've already assigned you to https://github.com/exercism/cairo/issues/180, once you tackle that, we can assign you to other tasks
